src/analysis/processing/qgsalgorithmspatialjoin.cpp

```diff
@@ -0,0 +1,375 @@
+/***************************************************************************
+                         qgsalgorithmspatialjoin.cpp
+                         ---------------------
+    begin                : November 2019
+    copyright            : (C) 2019 by
+    email                :
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmspatialjoin.h"
+#include "qgsprocessing.h"
+#include "qgsgeometryengine.h"
+#include "qgsvectorlayer.h"
+#include "qgsapplication.h"
+#include "qgsfeature.h"
+#include "qgsfeaturesource.h"
+
+///@cond PRIVATE
+
+
+void QgsSpatialJoinAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ),
+                QObject::tr( "Base Layer" ), QList< int > () << QgsProcessing::QgsProcessing::TypeVectorAnyGeometry ) );
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "JOIN" ),
+                QObject::tr( "Join Layer" ), QList< int > () << QgsProcessing::QgsProcessing::TypeVectorAnyGeometry ) );
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "PREDICATE" ), QObject::tr("Geometric predicate"),
+                tags(), true, 0 ) );
```

Comment:
```suggestion
                QStringList() << QObject::tr("intersects") << .... etc, true, 0 ) );
```

Reply:
This parameter needs the widget_wrapper metadata set from the original python implementation

Reply:
Should be done.

Reply:
Was that pushed yet?

Reply:
Yes, see line 37.